homeassistant/components/dnsip/config_flow.py

```diff
@@ -146,47 +146,48 @@ async def async_step_user(
         )
 
 
-class DnsIPOptionsFlowHandler(OptionsFlow):
+class DnsIPOptionsFlowHandler(OptionsFlowWithConfigEntry):
     """Handle a option config flow for dnsip integration."""
 
-    def __init__(self, entry: ConfigEntry) -> None:
-        """Initialize options flow."""
-        self.entry = entry
-
     async def async_step_init(
         self, user_input: dict[str, Any] | None = None
     ) -> ConfigFlowResult:
         """Manage the options."""
         errors = {}
         if user_input is not None:
+            user_input[CONF_RESOLVER] = user_input.get(CONF_RESOLVER, DEFAULT_RESOLVER)
+            user_input[CONF_RESOLVER_IPV6] = user_input.get(
+                CONF_RESOLVER_IPV6, DEFAULT_RESOLVER_IPV6
+            )
```

Comment:
```suggestion
            user_input.setdefault(CONF_RESOLVER, DEFAULT_RESOLVER)
            user_input.setdefault(CONF_RESOLVER_IPV6, DEFAULT_RESOLVER_IPV6)
```
Would this work here?

Reply:
No, as `user_input` can both contain these or not